### PR TITLE
Remove superfluous MediaRecorderErrorEvent interface

### LINKED ISSUE #211: MediaRecorderErrorEvent is a bad idea; use ErrorEvent instead
[ErrorEvent](https://html.spec.whatwg.org/#errorevent) already exists and serves this purpose fine. We shouldn't propagate multiple such types on the web platform.

### PR DESCRIPTION
**Pull request by dontcallmedom**
fixes #211

***
<a href="https://pr-preview.s3.amazonaws.com/dontcallmedom/mediacapture-record/pull/212.html" title="Last updated on Jan 20, 2022, 9:47 AM UTC (071eeac)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/mediacapture-record/212/3fd1025...dontcallmedom:071eeac.html" title="Last updated on Jan 20, 2022, 9:47 AM UTC (071eeac)">Diff</a>

### CODE REVIEW

File: MediaRecorder.bs [hunk 1]
```diff
@@ -670,7 +670,7 @@ dictionary BlobEventInit {
 
 The UA will throw a {{DOMException}} when the error can be detected at the time
 that the call is made. In all other cases the UA will <a>fire an event</a> named
-{{MediaRecorderErrorEvent}}.  If recording has been started and not yet stopped
+{{ErrorEvent}}.  If recording has been started and not yet stopped
```
Comment (alvestrand) [inline MediaRecorder.bs]: Is this an "event named ErrorEvent" or "an ErrorEvent"?

File: MediaRecorder.bs [hunk 2]
```diff
@@ -680,68 +680,27 @@ maximum {{Blob}} size that it will support, or the number of
 {{MediaStreamTrack}}s it will record at once.
 It will signal a fatal error if these limits are exceeded.
 
-## MediaRecorderErrorEvent ## {#errorevent-section}
-
-The {{MediaRecorderErrorEvent}} interface is defined for cases when an event is
-raised that was caused by an error.
+## Error eventsMediaRecorderErrorEvent ## {#errorevent-section}
```
Comment (alvestrand) [inline MediaRecorder.bs]: Title is mangled.

File: MediaRecorder.bs [hunk 3]
```diff
@@ -680,68 +680,27 @@ maximum {{Blob}} size that it will support, or the number of
...
 
 To <dfn>fire an error event</dfn>
```
Comment (alvestrand) [inline MediaRecorder.bs]: Shouldn't this whole section be replaced with a reference to https://html.spec.whatwg.org/multipage/webappapis.html#the-event-handler-processing-algorithm ?

Comment (dontcallmedom) [reply MediaRecorder.bs]: I've simplified to make "fire an error event" mean "fire an event with ErrorEvent as constructor" which I think achieves what we need

**Comment by foolip** [comment]
Ping reviewers on this. https://chromium-review.googlesource.com/c/chromium/src/+/3379631 is blocked wondering if this is going to land or not.

**Review by alvestrand — APPROVED** [review]
I approve of this simplification.

**Pull request merged**